File: buySellStock1.java
// Time Complexity : O(n) where n is the length of the array prices
// Space Complexity : O(1)
// Did this code successfully run on Leetcode : Yes
// Any problem you faced while coding this : No

// Your code here along with comments explaining your approach

class buySellStock1 {
    public int maxProfit(int[] prices) {
        int max = 0, min = Integer.MAX_VALUE;
        for (int price : prices) {
            min = Math.min(min, price);
            max = Math.max(max, price - min);
        }
        return max;
    }
}

File: buySellStock1.cpp
class Solution {
public:
    int maxProfit(vector<int>& prices) 
    {
        
        int n=prices.size();
        
        if(n==0)
            return 0;
        
        int min_prices=prices[0];
        int profit=0;
        for(int i=0;i<n;i++)
        {
            min_prices=min(min_prices,prices[i]);
            profit=max(profit,(prices[i]-min_prices));
        }
        
        return profit;
    }
};